### PR TITLE
Moved quoteWinArg, quotUnixArg, isEof out of StringTools to haxe.SysTools

### LINKED ISSUE #8109: [std] move arg quoting functions out of StringTools
It doesn't seem like the `StringTools` additions from https://github.com/HaxeFoundation/haxe/commit/767b3fb4021124276741473ee2185f70b8508a92  (`quoteUnixArg()`, `winMetaCharacters`, `quoteWinArg()`) really belong there. It's especially awkward to have those functions show up in String dot completion with `using StringTools`.

According to @ncannasse `StringTools` shouldn't even have a regex dependency... https://github.com/HaxeFoundation/haxe/pull/7608#issuecomment-480388097

I think long-term these should be moved elsewhere (`sys.SysTools` or something?). For now it seems sensible to at least annotate these with `@:noCompletion`.

### PR DESCRIPTION
**Pull request by RealyUniqueName**
`StringTools.isEof` -> `haxe.SysTools.isEofChar`
`StringTools.quoteWinArg` -> `haxe.SysTools.quoteWinArg`
`StringTools.quoteUnixArg` -> `haxe.SysTools.quoteUnixArg`

To keep backward compatibility methods in `StringTools` changed to invoke corresponding methods from `haxe.SysTools` with deprecation warnings. 

Closes #8109

### CODE REVIEW

File: std/StringTools.hx [hunk 1]
```diff
@@ -517,25 +504,16 @@ class StringTools {
...
-		return "'" + replace(argument, "'", "'\"'\"'") + "'";
+		return inline haxe.SysTools.quoteUnixArg(argument);
 	}
 
 	/**
 		Character codes of the characters that will be escaped by `quoteWinArg(_, true)`.
 	**/
 	@:noCompletion
+	@:deprecated('StringTools.winMetaCharacters is deprecated. Use haxe.SysTools.winMetaCharacters instead.')
 	public static var winMetaCharacters = [" ".code, "(".code, ")".code, "%".code, "!".code, "^".code, "\"".code, "<".code, ">".code, "&".code, "|".code, "\n".code, "\r".code, ",".code, ";".code];
```
Comment (Gama11) [inline std/StringTools.hx]: Shouldn't this just be a reference to the new array?

Comment (RealyUniqueName) [reply std/StringTools.hx]: The new field has different type and access.

Comment (Gama11) [reply std/StringTools.hx]: Only at compile time though.. `cast`? :)

Comment (Gama11) [reply std/StringTools.hx]: Or it could be turned into a `get, never` property that redirects.

Comment (RealyUniqueName) [reply std/StringTools.hx]: Changed to `cast SysTools.winMetaCharacters`

**Comment by Simn** [comment]
I'm a bit concerned about deprecation warnings about `isEof` because that's used in many parsers. 

Also the `fastCodeAt` documentation should be changed because it mentions `StringTools.isEof()` still.

**Comment by RealyUniqueName** [comment]
> I'm a bit concerned about deprecation warnings about isEof because that's used in many parsers.

That's why it's just deprecated. Not removed )

**Comment by Simn** [comment]
Yes but we still have to option to disable deprecation warnings. Maybe that's becoming a requirement for Haxe 4.

Anyway, I'm fine with making this change regardless.

**Comment by RealyUniqueName** [comment]
Maybe it's time to revive https://github.com/HaxeFoundation/haxe/pull/6725?

**Comment by Gama11** [comment]
There's `-D no-deprecation-warnings` at least, but that turns them all off of course.

I actually thought that @ncannasse [wanted `ifEof` to stay where it is](https://github.com/HaxeFoundation/haxe/issues/8109#issuecomment-481141084), but I may have misunderstood.

**Comment by RealyUniqueName** [comment]
> for backward compatibility

This PR is backward compatible )

**Comment by ncannasse** [comment]
isEof should be kept in StringTools, it's got nothing to do with Sys
actually, it's the "end of string" although named "eof", and is used
together with fastCodeAt.

**Comment by Gama11** [comment]
Should this be in the `haxe` or in the `sys` package then?

**Comment by ncannasse** [comment]
Let's keep it in StringTools with @:noUsing for now.

**Comment by Gama11** [comment]
I was talking about the location of the new `SysTools`.

**Comment by RealyUniqueName** [comment]
@ncannasse ^

**Comment by ncannasse** [comment]
Good for me

**Comment by RealyUniqueName** [comment]
CI failure is not related.

**Pull request merged**